### PR TITLE
New Presence BeatSaver

### PR DESCRIPTION
**Pull request by joerkig**
New presence for [BeatSaver](https://beatsaver.com)
2TtEZqRis4
Discord_5cRVGL9X3i
Discord_M67SrlzPZn
Discord_Qh8eu4OI4A
Discord_33TUMGXlPf
Discord_Iu4EsFffBi

### CODE REVIEW

File: websites/B/BeatSaver/presence.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  else if (document.location.pathname == ("/legal/dmca"))
+    presenceData.details = "Viewing DMCA Policy";
+  else if (document.location.pathname == ("/auth/login"))
+    presenceData.details = "Logging In...";
+  else if (document.location.pathname == ("/auth/register"))
+    presenceData.details = "Registering...";
+  else if (document.location.pathname == ("/user/upload"))
+    presenceData.details = "Uploading...";
+  else if (document.location.href == ("https://beatsaver.com/"))
+    presenceData.details = "Viewing Home Page";
```
Comment (ririxidev) [inline websites/B/BeatSaver/presence.ts:89]:
```suggestion
  else {
    switch(document.location.pathname) {
        case "/legal/dmca": /* code */
        case "/auth/login": /* code */
        case "/auth/register": /* code */
        case "/user/upload": /* code */
        case "/": /* code */
    }
  }
```

File: websites/B/BeatSaver/presence.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  if (!time)
+    delete presenceData.startTimestamp;
+
+  if (!buttons)
```
Comment (ririxidev) [inline websites/B/BeatSaver/presence.ts:94]:
```suggestion
  if (!buttons && presenceData.buttons)
```

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/B/BeatSaver/presence.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    case "/auth/register":
+      presenceData.details = "Registering...";
+      break;
+    case "/user/upload":
+      presenceData.details = "Uploading...";
+      break;
+    case "/":
+      presenceData.details = "Viewing Home Page";
+      break;
+  }
```
Comment (Bas950) [inline websites/B/BeatSaver/presence.ts:102]: I have a feeling this part is not working correctly, please double check and upload new proof of it working.

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/B/BeatSaver/presence.ts (hunk 3 above)
Comment (joerkig) [reply websites/B/BeatSaver/presence.ts:102]: New proof, Every page seems to be working
Discord_3jvtAJnHVc
Discord_3tlIdkOZZy
Discord_D6uuRVZsRM
Discord_Do4BAODXYw
Discord_FD9U5awLN7
Discord_IPWlRhgqdp
Discord_m9sSoBO0hX
Discord_SO047YfUPg
Discord_u4Kuwkdzbi
Discord_UdelIDqpoo
Discord_vDKJJemIGC
Discord_zPZyImOphl

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Pull request merged**